### PR TITLE
Favorites View Implementation

### PR DESCRIPTION
**Pull request by codingAlchemist**

### CODE REVIEW

**Comment by codingAlchemist** [comment]
This also has user story card #101679064 finished

File: OceanCandy/FavoritesViewController.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  FavoritesViewController.h
+//  OceanCandy
+//
+//  Created by Hashrocket Developer on 8/19/15.
+//  Copyright (c) 2015 jason debottis. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+@interface FavoritesViewController : UITableViewController
+@property (nonatomic, strong) NSMutableArray *favoriteStations;
```
Comment (greis) [inline OceanCandy/FavoritesViewController.h]: If you are not exposing this property to other classes it should be defined inside the private header in the implementation class.

File: OceanCandy/FavoritesViewController.m [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+- (void)viewWillAppear:(BOOL)animated{
+    [super viewWillAppear:animated];
+    NSArray *favorites = [Station MR_findAll];
+    self.favoriteStations = [[NSMutableArray alloc]initWithArray:[Station MR_findAll]];
```
Comment (greis) [inline OceanCandy/FavoritesViewController.m]: It looks like you are calling `[Station MR_findAll]` twice. There is no need for the `*favorites` variable

Comment (greis) [inline OceanCandy/FavoritesViewController.m]: Do you need to wrap around a mutable array? Or you can just use the array returned from MagicalRecord?

File: OceanCandy/StateListViewController.m [hunk 3]
```diff
@@ -41,8 +36,20 @@ - (void)viewDidLoad {
     }];
     self.stateListTable.delegate = self;
     self.stateListTable.dataSource = self;
+    
 }
+- (void)viewDidAppear:(BOOL)animated{
+    [super viewDidAppear:animated];
+    [self setTitle:@"States"];
+    NSArray *favorites = [Station MR_findAll];
```
Comment (greis) [inline OceanCandy/StateListViewController.m]: MagicalRecord has another method when you just want to count the entities: `[Station MR_numberOfEntities]`

File: OceanCandy/TideInfoViewController.m [hunk 4]
```diff
@@ -70,4 +116,9 @@ - (void)underlineTextLabels:(UILabel *)label{
     
 }
 
+- (IBAction)addToFavorites:(id)sender {
+    NSLog(@"station saved: %@", self.station.name);
+    NSLog(@"data %@",[self.data valueForKeyPath:@"High.feet"]);
+    [[StationService new]savefavorite:_station withResponse:self.data];
```
Comment (greis) [inline OceanCandy/TideInfoViewController.m]: Use `self.station` instead of `_station`

File: OceanCandy/StationService.m [hunk 5]
```diff
@@ -51,65 +49,56 @@ -(void)stations:(void (^)(NSArray *))callback {
...
-        }];
-        
-        
+    [self requestPath:path complete:^(id responseObject) {        
         callback(responseObject);
     }];
     
 }
-- (void)saveData{
+- (void)savefavorite:(Station *)station withResponse:(id)responseObject{
```
Comment (greis) [inline OceanCandy/StationService.m]: Always CamelCase: `saveFavorite`

File: OceanCandy/FavoritesViewController.m (hunk 2 above)
Comment (codingAlchemist) [inline OceanCandy/FavoritesViewController.m]: Other not needed array removed but the favorites are a mutable array to remove and add favorites.

File: OceanCandy/StateListViewController.m (hunk 3 above)
Comment (codingAlchemist) [inline OceanCandy/StateListViewController.m]: Thanks forgot about that method , but now i recall seeing it.

File: OceanCandy/TideInfoViewController.m (hunk 4 above)
Comment (codingAlchemist) [inline OceanCandy/TideInfoViewController.m]: done

File: OceanCandy/StationService.m (hunk 5 above)
Comment (codingAlchemist) [inline OceanCandy/StationService.m]: done

**Pull request closed without merging**